Add docs page about experimental features (#3942)
plone/volto: docs/source/configuration/experimental.md

---
myst:
  html_meta:
    "description": "Enable experimental features in Volto's configuration object"
    "property=og:description": "Enable experimental features in Volto's configuration object"
    "property=og:title": "Experimental features"

>     "keywords": "Volto, Plone, frontend, experimental"

Reviewer: ```suggestion
    "keywords": "Volto, Plone, frontend, experimental, features"
```

Author: @stevepiercy Sure, we can add this. Is it generally necessary to put words in keywords that are already in the title?

Reviewer: They are converted to meta tags. Just standard SEO stuff.

---

> # Experimental features

Reviewer: Is there a relation between "experimental features" and a "canary" release? Thinking out loud here... For example, does a canary release include experimental features? I happened to mention that in another PR:

https://github.com/plone/volto/pull/3934/files#diff-c060614717ad6cda93b44a2e34baa9d862e93884d0318798631e34808ff3842dR11-R13

Reviewer: Oh, also "feature flags". Does that belong in here?

Experimental features are planned for inclusion in a future release of Volto,
but are not yet considered mature by the community.

> 

Reviewer: ```suggestion
An experimental feature would typically be released in a canary release, giving users an easy way to install it and see if anything breaks before we make a new stable release that includes it.

```

Author: @stevepiercy I don't think this really helps clarify. That's true of non-experimental features too.

Reviewer: Maybe this?

```suggestion
An experimental feature gives users an easy way to install it and see if anything breaks before we make a new stable release that includes it.

```

## Volto configuration

> You need to enable experimental features in Volto's configuration object:

Reviewer: ```suggestion
You can enable an experimental feature—also called a "feature flag"—in Volto's configuration object as shown below.
```
